src/coredump/stacktrace.c

```diff
@@ -408,12 +416,16 @@ static int parse_core(int fd, const char *executable, char **ret, JsonVariant **
                 return -EINVAL;
         }
 
-        r = fflush_and_check(c.f);
-        if (r < 0)
-                return log_warning_errno(r, "Could not parse core file, flushing file buffer failed: %m");
+        if (c.f) {
+                r = fflush_and_check(c.f);
+                if (r < 0)
+                        return log_warning_errno(r, "Could not parse core file, flushing file buffer failed: %m");
+        }
 
-        c.f = safe_fclose(c.f);
-        *ret = TAKE_PTR(buf);
+        if (ret) {
```

Comment:
If `ret` is non-NULL, then `c.f` must be also non-NULL. So, how about the following?
```
if (ret) {
    r = fflush_and_check();
    if (r < 0)
        return log_warning_errno();

    c.f = safe_fclose(c.f);
    *ret = TAKE_PTR(buf);
}
```